src/Apps/Collect/Components/Collection/Header/DefaultHeader.tsx

```diff
@@ -0,0 +1,170 @@
+import { Box, color, Flex, Image } from "@artsy/palette"
+import { DefaultHeader_headerArtworks } from "__generated__/DefaultHeader_headerArtworks.graphql"
+import React, { FC, useEffect, useState } from "react"
+import { createFragmentContainer, graphql } from "react-relay"
+import styled from "styled-components"
+import { getViewportWidth } from "Utils/viewport"
+
+interface Props {
+  headerArtworks: DefaultHeader_headerArtworks
+  defaultHeaderImageHeight: number
+}
+
+const IMAGE_MARGIN_X = 10
+const LARGE_VIEWPORT_WIDTH = 880
+
+export const CollectionDefaultHeader: FC<Props> = ({
+  headerArtworks,
+  defaultHeaderImageHeight,
+}) => {
+  handleViewportResize()
+  const { hits: artworks } = headerArtworks
+
+  if (!artworks) {
+    return null
+  }
+
+  const viewportWidth = getViewportWidth()
+  const smallViewport = viewportWidth < LARGE_VIEWPORT_WIDTH
+  const duplicatedArtworks = artworks.slice(0)
+  const artworksToRender = getHeaderArtworks(
+    duplicatedArtworks,
+    viewportWidth,
+    smallViewport
+  )
+
+  return (
+    <header>
+      <DefaultHeaderContainer
+        position={["relative", "absolute"]}
+        left={["auto", 0]}
+        width={["auto", 1]}
+        height={[160, 160, 250]}
+      >
+        <HeaderArtworks>
+          {artworksToRender.map((artwork, i) => {
+            return (
+              <a href={artwork.href} key={i}>
+                <Image
+                  mx={0.5}
+                  height={defaultHeaderImageHeight}
+                  src={
+                    smallViewport
+                      ? artwork.image.small.url
+                      : artwork.image.large.url
+                  }
+                  preventRightClick
+                />
+              </a>
+            )
+          })}
+        </HeaderArtworks>
+      </DefaultHeaderContainer>
+    </header>
+  )
+}
+
+const handleViewportResize = () => {
```

Comment:
This feels like it would make a great custom hook! It could be named something like `useViewportSize` or `useWindowSize`, and it could go in our custom hooks folder (`src/utils/Hooks`).

Reply:
Ooh, good point! I'll add move this now.